### DISCUSSION TITLE
How to use Vercel's Image Optimization with Remix?

### DISCUSSION THREAD

**Discussion by jorisre** [vercel/vercel#8338] [Help] [answered]
Hi 👋🏻 

My Remix project is deployed on Vercel. I would like to use Vercel's Image Optimization with it. 

Unfortunately, it don't work.

Following the documentation and the example: https://vercel.com/docs/build-output-api/v3#build-output-configuration/supported-properties/images

I've manually populated the `.vercel/output` folder for testing purpose, once deployed _(without error)_, 
I'm getting a bad request with the url: `https://<MY_DOMAIN>.vercel.app/_vercel/image?url=/images/club-activity.jpg&w=400&q=75`

```json
// .vercel/output/config.json
{
  "version": 3,
  "regions": ["cdg"],
  "build": {
    "env": {
      "ENABLE_FILE_SYSTEM_API": "1",
    },
  },
  "images": {
    "domains": [],
    "sizes": [640, 750, 828, 1080, 1200],
    "minimumCacheTTL": 60,
    "formats": ["image/webp", "image/avif"],
  },
}
```

```
400: BAD_REQUEST
Code: INVALID_IMAGE_OPTIMIZE_REQUEST
ID: cdg1::xxx
```

What's wrong? Thank for help ✌🏻

**Comment by amyegan** [comment]
Hi @jorisre! Sorry for the late response. Are you still seeing this error?

**Comment by TooTallNate** [comment, marked as answer]
You should put the `"images"` configuration in the `vercel.json` file:

```json
{
  "images": {
    "domains": [],
    "sizes": [640, 750, 828, 1080, 1200],
    "minimumCacheTTL": 60,
    "formats": ["image/webp", "image/avif"],
  }
}
```

I think you were getting that `INVALID_IMAGE_OPTIMIZE_REQUEST` error because of the `w=400` parameter, which is not a size listed in the `"sizes"` array in the configuration.

**Reply by WildEgo** [reply]
Hey sorry to bother u mate, the [schema](https://openapi.vercel.sh/vercel.json) doesn't seem to show this, adding "images" straight up breaks the deployment.

**Reply by TooTallNate** [reply]
Good catch, we'll get it added to the schema. How exactly is it breaking the deployment though? Can you share an example deployment URL?

**Reply by WildEgo** [reply]
I'm not fully sure but the Vercel bot just seems to fail as soon as I add the following
```
  images: {
    sizes: [
      640,
      750,
      828,
      1080,
      1200
    ],
    domains: [],
    minimumCacheTTL: 60,
    formats: [
      "image/avif",
      "image/webp"
    ],
    remotePatterns: [
      {
        protocol: "https",
        hostname: "project.gg",
        pathname: "**"
      },
      {
        protocol: "https",
        hostname: "api.project.gg",
        pathname: "**"
      }
    ]
  },

```

**Reply by TooTallNate** [reply]
@WildEgo Your issue has now been fixed, and `images` is also now present on the `vercel.json` schema file.
